### PR TITLE
JAPI-147 Centralize remote test connection options gathering

### PR DESCRIPTION
**Pull request by rpastrana**
- Provides baseremotetest abstract class
- Remote tests now extend baseremotest
- Connection to hpcc defaults to localhost:8010 no creds
- Connection and creds provided via java env props
-- Via Maven: mvn -DargLine="-Dhpccconnect=http://ip:8010 -Dhpccuser=user -Dhpccpass=pass -Dlzname=someip" install
-- Via Eclipse: VM Args in run profile: -Dhpccconnect=http://ip:8010 -Dhpccuser=user -Dhpccpass=pass -Dlzname=someip

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

**Comment by richardkchapman** [comment]
typo (gatering) in commit message

**Comment by jpmcmu** [comment]
@rpastrana looks good

File: wsclient/src/test/java/org/hpccsystems/ws/client/BaseWsAttributesClientIntegrationTest.java [hunk 1]
```diff
@@ -91,28 +68,28 @@ public void testWriteAttributes() throws Exception
                     new String(Files.readAllBytes(testdir.listFiles()[i].toPath()))));
         }
 
-        List<ECLAttributeInfo> found=client.findItems(testmodname, null,null,null,null,null);
+        List<ECLAttributeInfo> found=wsattclient.findItems(testmodname, null,null,null,null,null);
         if (found.size()>0)
```
Comment (drealeed) [inline wsclient/src/test/java/org/hpccsystems/ws/client/BaseWsAttributesClientIntegrationTest.java]: how will this test function when run against clusters like http://10.173.147.1:8010 that have no wsattributes service on port 8145? Is there a check somewhere to bypass this test?

Comment (rpastrana) [reply wsclient/src/test/java/org/hpccsystems/ws/client/BaseWsAttributesClientIntegrationTest.java]: @drealeed take a look at the latest commit. All tests targeting non eclwatch services will need to query the dedicated port for that service.
If this test shouldn't execute with its group  (IntegrationTest), we might need to create a new (ExtendedIntegrationTest) or something similar to that.

**Pull request merged**